scipy/stats/tests/test_stats.py

```diff
@@ -74,15 +75,38 @@
 TINY = array([1e-12,2e-12,3e-12,4e-12,5e-12,6e-12,7e-12,8e-12,9e-12], float)
 ROUND = array([0.5,1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5], float)
 
+lazy_xp_modules = [stats]
+lazy_xp_function(stats.tmean, static_argnames=("inclusive", "axis"))
+lazy_xp_function(stats.tvar, static_argnames=("inclusive", "axis", "ddof"))
+lazy_xp_function(stats.tstd, static_argnames=("inclusive", "axis", "ddof"))
+lazy_xp_function(stats.tsem, static_argnames=("inclusive", "axis", "ddof"))
+lazy_xp_function(stats.tmin, static_argnames=("inclusive", "axis"))
+lazy_xp_function(stats.tmax, static_argnames=("inclusive", "axis"))
+
+
+def eager_warns(x, warning_type, match=None):
+    """pytest.warns context manager, but only if x is not a lazy array."""
+    if is_lazy_array(x):
+        return contextlib.nullcontext()
+    return pytest.warns(warning_type, match=match)
+
 
 class TestTrimmedStats:
     # TODO: write these tests to handle missing values properly
     dprec = np.finfo(np.float64).precision
 
+    @staticmethod
+    def lazy_expect(expect, xp):
+        """The output dtype of tmin/tmax is data-dependent: int inputs are cast to
+        float if there are any NaNs. On lazy backends (JAX/Dask), since content tests
+        are impossible, int inputs are always cast to float.
+        """
+        expect = xp.asarray(expect)
+        if is_lazy_array(expect) and xp.isdtype(expect, "integral"):
+            return xp.astype(expect, xp.float64)
+        return expect
```

Comment:
What do you think about changing `test_tmin` and `test_tmax` to work with floating point input and adding one short, separate test that confirms that integer input still works? Then, I think this logic in this function could go right in the new test, instead.

Reply:
This is a great idea